### PR TITLE
fix(table): avoid unnecessary re-rendering if TableRowMenu doesn't exist

### PR DESCRIPTION
**Pull request by rivka-ungar**
https://monday.monday.com/boards/3532714909/views/80492480/pulses/7889513842

### CODE REVIEW

File: packages/core/src/components/Table/context/TableRowMenuContext/TableRowMenuContext.tsx [hunk 1]
```diff
@@ -15,12 +16,18 @@ export const TableRowMenuProvider = ({ value, children }: TableRowMenuProviderPr
...
     (rowRef: React.RefObject<HTMLDivElement>) => {
-      if (isMenuOpen || hoveredRowRef?.current === rowRef.current) return;
+      if (hasMenuRef.current === false || isMenuOpen || hoveredRowRef?.current === rowRef.current) return;
+
       resetTimeout();
       setHoveredRowRef(rowRef);
       const tableRootTop = tableRootRef.current.getBoundingClientRect().top;
       const rowTop = rowRef.current.getBoundingClientRect().top;
       setMenuButtonPosition(rowTop - tableRootTop);
+      if (hasMenuRef.current === null) {
```
Comment (YossiSaadi) [inline packages/core/src/components/Table/context/TableRowMenuContext/TableRowMenuContext.tsx:26]: || true?

Comment (YossiSaadi) [reply packages/core/src/components/Table/context/TableRowMenuContext/TableRowMenuContext.tsx:26]: or even
```suggestion
      if (hasMenuRef.current !== false) {
```

Comment (YossiSaadi) [reply packages/core/src/components/Table/context/TableRowMenuContext/TableRowMenuContext.tsx:26]: to avoid re-entering this if

Comment (rivka-ungar) [reply packages/core/src/components/Table/context/TableRowMenuContext/TableRowMenuContext.tsx:26]: Only the first time it is null. After that it is set to true/false. So checking if it is null will run only the first time. If I check not false than the check will run every time for the tables that don't have a menu

Comment (YossiSaadi) [reply packages/core/src/components/Table/context/TableRowMenuContext/TableRowMenuContext.tsx:26]: ohhh you're right actually
sorry :)

**Review by YossiSaadi — APPROVED** [review]
Awesome work!

**Pull request merged**